### PR TITLE
helm: publish 1.24.1 chart with new releases

### PR DESCRIPTION
**Pull request by jackfrancis**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR publishes a new 1.24.1 version of the cloud-provider-azure helm chart to reference the newest releases:

- https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.24.1
- https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.23.13
- https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.1.16
- https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.0.20

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: helm/repo/index.yaml [hunk 1]
```diff
@@ -20,34 +20,13 @@ entries:
...
-    - email: jackfrancis@gmail.com
-      name: Jack Francis
-    - email: zhechengli1995@outlook.com
-      name: Zhecheng Li
-    name: cloud-provider-azure
-    sources:
-    - https://github.com/kubernetes-sigs/cloud-provider-azure
-    urls:
-    - cloud-provider-azure-1.24.0+20220526.tgz
-    version: 1.24.0+20220526
```
Comment (lzhecheng) [inline helm/repo/index.yaml:45]: Maybe we can manually keep 1.24.0, 1.24.0+20220526?

Comment (jackfrancis) [reply helm/repo/index.yaml]: Agreed, done.

**Pull request merged**